Packs/GsuiteAuditor/Integrations/GsuiteAuditor/GsuiteAuditor.py

```diff
@@ -0,0 +1,344 @@
+    'CUSTOM_USER_SCHEMA_FIELD_DETAILS': 'Field Details',
+    'CUSTOM_USER_SCHEMA_UPDATE': 'Updated Custom User Schema Details',
+    'DATATRANSFER_REQUEST_CREATE_SUCCESS': 'Data Transfer Details',
+    'NOT_FOUND': 'No {} found.',
+    'USER_DELETE': 'User with user key {} deleted successfully.',
+    'USER_UPDATE': 'Updated User Details'
+}
+DATE_FORMAT = '%Y-%m-%dT%H:%M:%SZ'  # ISO8601 format with UTC, default in XSOAR
+
+''' HELPER FUNCTIONS '''
+
+
+def is_email_valid(email: str) -> bool:
+    """
+    Validates provided email is valid or not.
+
+    :param email: email string.
+    :return:  True if email is in valid format.
+    """
+
+    return True if re.match(emailRegex, email) else False
+
+
+def prepare_args_for_activities_list(args: Dict[str, str]) -> Dict[str, str]:
+    """
+    Prepares arguments for gsuite-activity-search command.
+
+    :param args: Command arguments.
+
+    :return: Prepared arguments.
+    """
+
+    return GSuiteClient.remove_empty_entities({
+        'eventName': args.get('event_name'),
+        'filters': args.get('filters'),
+        'orgUnitId': args.get('org_unit_id'),
+        'groupIdFilter': args.get('group_id'),
+        'actorIpAddress': args.get('actor_ip_address'),
+        'startTime': args.get('start_time'),
+        'endTime': args.get('end_time'),
+        'maxResults': GSuiteClient.validate_get_int(args.get('max_results'),
+                                                    MESSAGES['INTEGER_ERROR'].format('max_results')),
+        'pageToken': args.get('page_token')
+    })
+
+
+def prepare_readable_items(items: List[Dict[str, Any]]) -> List[Dict[str, Any]]:
+    readable_items = [
+        {
+            'time': item.get('id', {}).get('time'),
+            'applicationName': item.get('id', {}).get('applicationName'),
+            'email': item.get('actor', {}).get('email'),
+            'profileId': item.get('actor', {}).get('profileId'),
+            'ipAddress': item.get('ipAddress', ''),
+            'events': item['events']
+        } for item in items
+    ]
+
+    return GSuiteClient.remove_empty_entities(readable_items)
+
+
+def prepare_output_for_activities_list(response: Dict[str, Any]) -> Dict[str, Any]:
+    """
+        prepares context output for gsuite-activity-search.
+
+        :param response: API response.
+
+        :return: output dictionary.
+        """
+    output_items = [{'id': item['id'],
+                     'actor': item['actor'],
+                     'ipAddress': item.get('ipAddress', []),
+                     'events': item['events']} for item in response.get('items', [])]
+
+    return {
+        'GSuite.ActivitySearch': GSuiteClient.remove_empty_entities(output_items),
+        'GSuite.PageToken.ActivitySearch': {
+            'nextPageToken': response['nextPageToken']
+        } if response.get('nextPageToken', '') else {}
+    }
+
+
+def create_end_time(start_time: str, added_time: str) -> str:
```

Comment:
this seems good as long as activities can't be entered retroactively - if they can then each application should have its own end time. - We can ask the SOC what is the correct behavior here.